actors/builtin/market/deal.go

```diff
@@ -12,6 +22,147 @@ import (
 //}
 var PieceCIDPrefix = market0.PieceCIDPrefix
 
+// The DealLabel is a "kinded union" -- can either be a String or a byte slice, but not both
+// TODO: What represents the empty label and how do we marshall it?
```

Comment:
I'd either put these behind pointers, or store some kind of empty/string/bytes "tag" field.